components/SelectComponent.tsx

```diff
@@ -0,0 +1,79 @@
+import { fhirJson } from "@fhir-typescript/r4-core";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { Autocomplete } from "@mantine/core";
+
+/**
+ * interface for the props that a ResourceCodeEditor component takes in.
+ * @resourceType is a string containing the resourceType
+ * @setValue is a state variable function that is used to pass the code editor's current contents to a parent component
+ * @value is a state variable string that is used to update the value of the select component textbox
+ */
+export interface SelectComponentProps {
+  resourceType: string;
+  setValue: Dispatch<SetStateAction<string>>;
+  value: string;
+}
+
+/**
+ * @param props include the type interface SelectComponentProps
+ * @returns a component with a loading component, server error, or autocomplete select component populated with resource IDs
+ */
+export default function SelectComponent(props: SelectComponentProps) {
+  const resourceType = props.resourceType;
+  const setValue = props.setValue;
+  const value = props.value;
```

Comment:
Coming in hot: if you're gonna do this, might as well do this:

```suggestion
export default function SelectComponent({ resourceType, setValue, value }: SelectComponentProps) {
```

Read more about detructuring here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment